lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2171,16 +2149,93 @@ SynthesizeMainFunctionRequest::evaluate(Evaluator &evaluator,
   // mainType.main() from the entry point, and that would require fully
   // type-checking the call to mainType.main().
 
-  constraints::ConstraintSystemOptions lookupOptions;
-  if (context.LangOpts.EnableAsyncMainResolution)
-    lookupOptions |=
-        constraints::ConstraintSystemFlags::ConsiderNominalTypeContextsAsync;
+  constraints::ConstraintSystem CS(
```

Comment:
Nit: add `using namespace constraints;` right before this so you don't have to use `constraints::` everywhere.

Reply:
Done